be/src/exec/vectorized/olap_scan_node.cpp

```diff
@@ -263,6 +263,10 @@ void OlapScanNode::_scanner_thread(TabletScanner* scanner) {
             delete chunk;
             break;
         }
+        // Improve for select * from table limit x;
+        if (limit() != -1 && scanner->raw_rows_read() >= limit()) {
```

Comment:
is that correct? `raw_rows_read`? is there any possible predicates like `select * from t where t.id > 10 limit 10`

Reply:
Fixed